schema.py

```diff
@@ -117,6 +124,8 @@ def validate(self, data):
             coverage = set()  # matched schema keys
             # for each key and value find a schema entry matching them, if any
             sorted_skeys = sorted(s, key=priority)
+            # resort sorted_skeys with priority_of_dict_key
+            sorted_skeys = sorted(sorted_skeys, key=priority_of_dict_key)
```

Comment:
Well, many thanks for your review and proposal.

-  I'll update the gitignore file to ensure that files ignored before will continue to be ignored.
-  Add a static method in `Schema` instead of a top-level function, though it's hacky.